website/blog/2019-11-04-binding-react-navigation-5-to-ui-kitten.md

```diff
@@ -0,0 +1,429 @@
+
+In app.navigator.tsx file and replace the placeholder screen with Auth Navigator. This will make authentication screens to be the starter point of your app.
+
+```js
+import React from 'react';
+import { NavigationNativeContainer } from '@react-navigation/native';
+import { createStackNavigator } from '@react-navigation/stack';
+import { AuthNavigator } from './auth.navigator';
+import { AppRoute } from './app-routes';
+
+const Stack = createStackNavigator();
+
+export const AppNavigator = (config): React.ReactElement => (
+  <NavigationNativeContainer>
+    <Stack.Navigator headerMode='none'>
+      <Stack.Screen name={AppRoute.AUTH} component={AuthNavigator}/>
+    </Stack.Navigator>
+  </NavigationNativeContainer>
+);
+```
+
+<img src="assets/binding-react-navigation-5-to-ui-kitten/stack-navigator.gif" height="420" />
+
+### Step 2. Top tabs
+
+Say, our app has both in-progress and finished tasks. So, you should separate them to avoid a mess. Here you can make it with two tabs on the home screen. To do this, we need to have three screens: two for tabs and one master screen for navigation management between tabs. Unlike the Stack Navigator component, the Top Tabs `Navigator` has a special prop for the component to control navigation between tabs - `tabBar`. We will use it to configure the tab bar with UI Kitten components.
+
+Open `./src/navigators/todo.navigator.tsx` file and paste the following code:
+
+```js
+import React from 'react';
+import { createStackNavigator } from '@react-navigation/stack';
+import { createMaterialTopTabNavigator } from '@react-navigation/material-top-tabs';
+import { TodoScreen, TodoInProgressScreen, TodoDoneScreen } from '@app-scenes/todo';
+import { AppRoute } from './app-routes';
+
+const Stack = createStackNavigator();
+const TopTab = createMaterialTopTabNavigator();
+
+export const TodoNavigator = (): React.ReactElement => (
+  <TopTab.Navigator tabBar={TodoScreen}>
+    <TopTab.Screen name={AppRoute.TODO_IN_PROGRESS} component={TodoInProgressScreen}/>
+    <TopTab.Screen name={AppRoute.TODO_DONE} component={TodoDoneScreen}/>
+  </TopTab.Navigator>
+);
+```
+
+The code above will enable you to navigate with gestures between `In Progress` screen and `Done` screen, but not set up the Tab Bar. Open `./src/scenes/todo/todo.component.tsx` file and paste the following code:
+
+```js
+import React from 'react';
+import { TabBar, Tab, Divider } from 'react-native-ui-kitten';
+import { SafeAreaLayout, SaveAreaInset, SafeAreaLayoutElement } from '@app-components/safe-area-layout.component';
+import { Toolbar } from '@app-components/toolbar.component';
+import { DoneAllIcon, GridIcon } from '@app-assets/icons';
+
+export const TodoScreen = (props): SafeAreaLayoutElement => {
+
+  const onTabSelect = (index: number): void => {
+    const { [index]: selectedTabRoute } = props.state.routeNames;
+    props.navigation.navigate(selectedTabRoute);
```

Comment:
Instead of navigating manually, you can use the `onTabPress` prop passed to the tab bar. This makes sure events such as `tabPress` etc. get triggered.

```js
onTabPress({ route: state.routes[index] });
```

Reply:
Thanks for the explanation 👍 
Unfortunately, I've tried using this function, but it throws ts errors since it accepts the `route` and `preventDefault` function. See complete log on a [screenshot](https://monosnap.com/file/NOyY03FoyiO0E5QEUR8xLPNYsCwnqg)